middleware/http/server.go

```diff
@@ -101,6 +110,17 @@ func (h handler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	// try to extract B3 Headers from upstream
 	sc := h.tracer.Extract(b3.ExtractHTTP(r))
 
+	// store whitelisted headers to be propagated in spanContext
```

Comment:
I know you know what you are doing, but I've seen this copy pasted. is it worthwhile to have a utility yet? wonder if someone would forget to call Init() and possibly be inconsistent about the read, check length, copy-in thing